query-engine/query-structure/src/query_arguments.rs

```diff
@@ -28,6 +28,49 @@ pub struct QueryArguments {
     pub relation_load_strategy: Option<RelationLoadStrategy>,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum Take {
+    All,
+    One,
+    Some(i64),
+}
+
+impl Take {
+    pub fn is_all(self) -> bool {
+        self == Take::All
+    }
+
+    pub fn is_some(self) -> bool {
+        matches!(self, Take::Some(_))
+    }
+
+    pub fn abs(self) -> Option<i64> {
+        match self {
+            Take::All => None,
+            Take::One => Some(1),
+            Take::Some(n) => Some(if n < 0 { -n } else { n }),
```

Comment:
```suggestion
            Take::Some(n) => Some(n.abs()),
```

Reply:
It is strange. I've tried `n.abs()` and it failed to compile yesterday. Now it works with the current code. Must have overlooked something...